### PR TITLE
Deprecated calls of cltest and next portion of mockery usage fixes

### PR DESCRIPTION
**Pull request by pinebit**
Part of https://app.shortcut.com/chainlinklabs/story/44670/spike-mockery-usage-overhaul

### CODE REVIEW

File: core/internal/cltest/cltest.go [hunk 1]
```diff
@@ -470,25 +470,7 @@ func NewEthMocksWithDefaultChain(t testing.TB) (c *evmMocks.Client) {
...
-
-// Deprecated: use evmtest.NewEthClientMock
-func NewEthClientMock(t *testing.T) *evmMocks.Client {
-	return evmtest.NewEthClientMock(t)
-}
-
-// Deprecated: use evmtest.NewEthClientMockWithDefaultChain
-func NewEthClientMockWithDefaultChain(t *testing.T) *evmMocks.Client {
-	return evmtest.NewEthClientMockWithDefaultChain(t)
+	return evmMocks.NewClient(t)
```
Comment (jmank88) [inline core/internal/cltest/cltest.go:473]: :fire:

File: core/internal/testutils/evmtest/evmtest.go [hunk 2]
```diff
@@ -307,6 +308,7 @@ func (m *MockEth) AssertExpectations(t *testing.T) {
 		sub.AssertExpectations(t)
 	}
 }
+*/
```
Comment (jmank88) [inline core/internal/testutils/evmtest/evmtest.go:311]: Remove the whole block?

Comment (pinebit) [reply core/internal/testutils/evmtest/evmtest.go]: Thank you! Removed.

**Review by jmank88 — APPROVED** [review]
:partying_face: Less it more!
Screenshot from 2022-08-03 09-02-10

**Review by essamhassan — APPROVED** [review]
LGTM

**Pull request merged**